src/api/firebase.js

```diff
@@ -63,14 +69,37 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
+export async function updateItem(listId, itemData) {
+	const docRef = doc(db, listId, itemData.id);
+
+	await updateDoc(docRef, {
+		isChecked: !itemData.isChecked,
+		dateLastPurchased: new Date(Date.now()),
+		totalPurchases: itemData.totalPurchases,
```

Comment:
here you should use the data that you passed from where the function was used

Reply:
That was the initial thing I was passing as item.dateLastPurchased and you requested a change